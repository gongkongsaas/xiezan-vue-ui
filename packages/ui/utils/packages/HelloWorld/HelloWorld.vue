<template>
  <div class="demo-box">
    <h1>hello {{ msg }}</h1>
    <button @click="count++">count is: {{ count }}</button>
    <p>
      Edit <code>components/HelloWorld.vue</code> to test hot module
      replacement.
    </p>
  </div>
</template>

<script lang="ts">
import { defineComponent, ref } from 'vue'
export default defineComponent({
  name: "HelloWorld",
  props: {
    msg: String,
  },
  data() {
    return {
      count: 0,
    };
  },
});
</script>
<style scoped>
.demo-box {
  font-size: 24px;
  font-weight: bold;
}
</style>
