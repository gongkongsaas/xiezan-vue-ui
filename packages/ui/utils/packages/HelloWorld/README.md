---
title: Hello-world
description: Hello-world component doc
tags:
  - "components"
---

# Hello-world

this is Hello-world component

### Basic

description

> show demo

```vue
<template>
  <hello-world :msg="msg"></hello-world>
</template>
<script>
export default {
  data() {
    return {
      msg: "world",
    };
  },
};
</script>
```
