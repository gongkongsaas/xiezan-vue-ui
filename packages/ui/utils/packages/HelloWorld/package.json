{
  "name": "@vitec/hello-world",
  "version": "0.0.1",
  "main": "lib/index.js",
  "dependencies": {
    "vue": "^3.0.0-rc.1"
  }
}
