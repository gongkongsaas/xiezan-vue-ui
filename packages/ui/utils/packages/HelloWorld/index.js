import HelloWorld from "./HelloWorld.vue";

const install = function (Vue, opts = {}) {
  Vue.component(HelloWorld.name, HelloWorld);
};

export default { install, HelloWorld };
