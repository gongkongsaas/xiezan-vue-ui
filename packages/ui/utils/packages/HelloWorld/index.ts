import { App } from "vue";
import HelloWorld from "./HelloWorld.vue";

export default (app: App): void => {
  app.component(HelloWorld.name, HelloWorld);
};
