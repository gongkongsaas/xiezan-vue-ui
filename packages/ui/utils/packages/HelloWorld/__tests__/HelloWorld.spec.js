import { mount } from "@vue/test-utils";
import { defineComponent } from "vue";
import HelloWorld from "../HelloWorld.vue";

const compC = defineComponent({
  template: '<hello-world msg="message"></hello-world>',
  components: { HelloWorld },
});

describe("element", () => {
  it("find h1", () => {
    const wrapper = mount(compC);

    expect(wrapper.element.querySelector('h1').innerHTML).toContain('message')
  });
  it("snppshot match", () => {
    const wrapper = mount(compC);

    expect(wrapper.element.innerHTML).toMatchSnapshot()
  });
});
