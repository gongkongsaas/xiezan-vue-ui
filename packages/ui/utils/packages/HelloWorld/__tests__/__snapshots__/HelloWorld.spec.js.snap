// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`element snppshot match 1`] = `"<h1>hello message</h1><button>count is: 0</button><p> Edit <code>components/HelloWorld.vue</code> to test hot module replacement. </p>"`;
